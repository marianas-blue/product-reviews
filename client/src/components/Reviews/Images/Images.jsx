/* eslint-disable react/no-access-state-in-setstate */
/* eslint-disable react/destructuring-assignment */
/* eslint-disable import/extensions */
/* eslint-disable react/no-unused-state */
/* eslint-disable no-useless-constructor */
/* eslint-disable react/prefer-stateless-function */
import React from 'react';
import { Title, Pic, SeeImages } from './Images.style';
import Modal from '../Modal/Modal.jsx';

const axios = require('axios');

class Images extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      images: [],
      isOpen: false,
      refresh: false,
    };
    this.toggleModal = this.toggleModal.bind(this);
  }

  componentDidMount() {
    const fullUrl = document.URL;
    const urlArray = fullUrl.split('/');
    const lastSegment = urlArray[urlArray.length - 1];
    axios.get(`http://ec2-34-211-238-169.us-west-2.compute.amazonaws.com/api/product/${lastSegment}`)
      .then((res) => {
        for (let i = 0; i < res.data[0].images.length; i += 1) {
          this.state.images.push(<img src={res.data[0].images[i]} height="120" width="120" alt="test" key={i} style={{ margin: '2.5px' }} />);
        }
        this.setState({
          refresh: true,
        });
      });
  }

  toggleModal() {
    this.setState({
      isOpen: !this.state.isOpen,
    });
  }
 
  render() {
    return (
      <div>
        <Title>Customer Images</Title>
        <div id="app2">
          <div>
            {this.state.images}
          </div>
          <SeeImages onClick={this.toggleModal}>
            See all customer images
          </SeeImages>

          <Modal
            show={this.state.isOpen}
            onClose={this.toggleModal}
          >
            <Pic src="http://lorempixel.com/640/480/abstract" alt="abstract" height="120" width="120" />
            <Pic src="http://lorempixel.com/640/480/animals" alt="animals" height="120" width="120" />
            <Pic src="http://lorempixel.com/640/480/business" alt="business" height="120" width="120" />
            <Pic src="http://lorempixel.com/640/480/cats" alt="cats" height="120" width="120" />
            <Pic src="http://lorempixel.com/640/480/city" alt="city" height="120" width="120" />
            <Pic src="http://lorempixel.com/640/480/food" alt="food" height="120" width="120" />
            <Pic src="http://lorempixel.com/640/480/nightlife" alt="nightlife" height="120" width="120" />
            <Pic src="http://lorempixel.com/640/480/fashion" alt="fashion" height="120" width="120" />
            <Pic src="http://lorempixel.com/640/480/people" alt="people" height="120" width="120" />
            <Pic src="http://lorempixel.com/640/480/nature" alt="nature" height="120" width="120" />
            <Pic src="http://lorempixel.com/640/480/sports" alt="sports" height="120" width="120" />
            <Pic src="http://lorempixel.com/640/480/technics" alt="technics" height="120" width="120" />
            <Pic src="http://lorempixel.com/640/480/transport" alt="transport" height="120" width="120" />
            <Pic src="https://picsum.photos/120/120/?random" alt="random" />
            <Pic src="https://picsum.photos/120/120?image=1080" alt="fruit" />
          </Modal>
        </div>
      </div>
    );
  }
}

export default Images;
