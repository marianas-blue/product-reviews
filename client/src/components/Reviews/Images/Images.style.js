import styled from 'styled-components';

export const Title = styled.button`
font-weight: bolder;
font-size: 17px;
line-height: 1.255;
border: none; 
padding: 0;
background-color: transparent;
outline: none;
text-align: left;
`;

export const Pic = styled.img`
  margin: 2.5px;
`;

export const SeeImages = styled.button`
  color: #0066c0;
  font-size: 13px 
  text-Decoration: none;
  :hover {
    text-decoration: underline;
    color: #c60;
  }
  border: none;
  outline: none;
  padding: 0;
`;
