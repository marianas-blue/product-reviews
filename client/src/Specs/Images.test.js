import React from 'react';
import { shallow } from 'enzyme';

import Images from '../components/Reviews/Images/Images';

// eslint-disable-next-line no-undef
describe('Images', () => {
  // eslint-disable-next-line no-undef
  it('should render correctly', () => {
    // eslint-disable-next-line react/jsx-filename-extension
    const component = shallow(<Images />);

    // eslint-disable-next-line no-undef
    expect(component).toMatchSnapshot();
  });
});
