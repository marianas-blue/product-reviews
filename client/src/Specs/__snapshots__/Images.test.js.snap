// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Images should render correctly 1`] = `
<div>
  <styled.button>
    Customer Images
  </styled.button>
  <div>
    <styled.img
      alt="food"
      height="144"
      src="http://lorempixel.com/640/480/food"
      width="144"
    />
    <styled.img
      alt="cat"
      height="144"
      src="http://lorempixel.com/640/480/cats"
      width="144"
    />
    <styled.img
      alt="sports"
      height="144"
      src="http://lorempixel.com/640/480/sports"
      width="144"
    />
    <styled.img
      alt="transportation"
      height="144"
      src="http://lorempixel.com/640/480/transport"
      width="144"
    />
  </div>
  <styled.a
    href="#top"
  >
    See all customer images
  </styled.a>
</div>
`;
